src/httpServer/middleware.ts

```diff
@@ -84,3 +86,35 @@ export function KoaMiddleware(...koaMiddleware: Koa.Middleware[]) {
   }
   return clsdec;
 }
+
+/////////////////////////////////
+/* OPEN API DECORATORS */
+/////////////////////////////////
+
+// Note, OAuth2 is not supported yet.
+type SecurityScheme = Exclude<OpenApi3.SecuritySchemeObject, OpenApi3.OAuth2SecurityScheme>;
+
+/**
+ * Declare an OpenApi Security Scheme (https://spec.openapis.org/oas/v3.0.3#security-scheme-object
+ * for the methods of a class. Note, this decorator is only used in OpenApi generation and does not
+ * affect runtime behavior of the Operon app.
+ */
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export function OpenApiSecurityScheme(securityScheme: SecurityScheme) {
+  return function <T extends { new(...args: unknown[]): object }>(_ctor: T) { }
+}
+
+/**
+ * Declare that the decorated method should not declare a security requirement.
+ * By default, all methods in a class decorated with @OpenApiSecurityScheme specify
+ * the named security requirement associated with this class.
+ * This decorator overrides that behavior for the decorated method.
+ */
+
+export function OpenApiAnonymous() {
```

Comment:
On a second thought, I think we should have an `@Anonymous()` decorator no matter what. Currently, we apply Koa middleware to all endpoints in a class, for example the Koa-JWT verification one. Then the issue is that all routes would require an auth token.

As you said, we don't have a good way to differentiate "any authenticated users" (empty required roles) vs. "anonymous requests" (no auth required). So I wonder if we can update this decorator to an actual Operon decorator that will affect runtime behavior (maybe in a separate PR)?

Reply:
How about I rename it to just `@Anonymous` for this PR and tackle runtime integration in a future PR?

Reply:
BTW, with this change, it seems *more* reasonable that the OpenAPI declarative info should be an optional parameter to `@Authentication`

Reply:
I have a local change prepped for `OpenApiAnonymous` -> `Anonymous`. Holding off for @qianl15 response to above question.

Reply:
As per a [separate comment on this PR](https://github.com/dbos-inc/operon/pull/163#issuecomment-1795488812), I'm removing `OpenApiAnonymous` completely. Our authz system currently doesn't look at authenticated